File: BAEKJOON/cpp/3181_줄임말_만들기.cpp
#include <iostream>
#include <vector>
#include <string>
#include <set>
#include <cctype>
using namespace std;

string s, answer;
set<string> st;
vector<string> words;
string ignore[10] = { "i", "pa", "te", "ni", "niti", "a", "ali",  "nego", "no", "ili" };


void input() {
	getline(cin, s);
}

void solution() {
	string temp;

	for (int i = 0; i < s.size(); i++) {
		if (s[i] == ' ') {
			words.push_back(temp);
			temp.clear();
			continue;
		}

		temp += s[i];
	}

	if (!temp.empty()) {
		words.push_back(temp);
		temp.clear();
	}

	for (int i = 0; i < 10; i++) {
		st.insert(ignore[i]);
	}

	for (int i = 0; i < words.size(); i++) {
		if ((i == 0 && st.find(words[i]) != st.end()) || (st.find(words[i]) == st.end())) {
			answer += toupper(words[i][0]);
		}
	}

	cout << answer;
}

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	input();
	solution();

	return 0;
}

File: BAEKJOON/java/3181_줄임말_만들기.java
import java.io.*;
import java.util.*;

public class Main {
    public static String s;
    public static StringBuilder answer = new StringBuilder();
    public static Set<String> set = new HashSet<>();

    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        input();
        solution();

        bw.flush();
        bw.close();
    }

    private static void input() throws IOException {
        s = br.readLine();
    }

    private static void solution() throws IOException {
        String[] ignore = { "i", "pa", "te", "ni", "niti", "a", "ali",  "nego", "no", "ili" };

        set.addAll(Arrays.asList(ignore));

        String[] words = s.split(" ");

        for (int i = 0; i < words.length; i++) {
            if (i == 0 || !set.contains(words[i])) {
                answer.append(words[i].toUpperCase().charAt(0));
            }
        }

        bw.write(answer.toString());
    }
}